### PR TITLE
LOG4J2-2973 Rename EventTemplateAdditionalField#type (conflicting with properties file parser) to #format

### PR DESCRIPTION
**Pull request by Fabricc**
[LOG4J2-2973](https://issues.apache.org/jira/browse/LOG4J2-2973) points to [a GitHub discussion](/apache/logging-log4j2/commit/61f125b8b879d1a0852b24358da7424baeb20c31#r44935541), where it has been shared that the injection of plug-in builder attributes of type array from properties files is not working. But... It actually works! The reason it doesn't work for injecting a field of type `EventTemplateAdditionalField[]` is the `EventTemplateAdditionalField#type` field whose name conflicts with the `type` keyword property file loader uses to determine the type of an element.

This PR renames `EventTemplateAdditionalField#type` to `format`. While this is **not** a backward compatible change, injection of JSON template layout `eventTemplateAdditionalFields` has already been broken in release 2.14.0 and the introduced fix has already broken the backward compatibility. See [LOG4J2-2961](https://issues.apache.org/jira/browse/LOG4J2-2961) for details. Hence, it should be okay. :crossed_fingers:

### CODE REVIEW

**Review by vy — COMMENTED** [review]
Great job @Fabricc! See the comments I have shared. I have some extra remarks/requests too:

1. `json-template-layout.adoc.vm` needs to be updated too. While updating it, would you also mind sharing a `.properties` file counterpart of the example at the bottom of the _Additional event template fields_ section, please?
2. Would you also update `changes.xml`, please? (I will rework the LOG4J2-2973 JIRA ticket title & description momentarily. You can copy paste from there.)

**Comment by jftai1** [comment]
Please  had my vote on fixing this issue...

I am hoping to use this feature in a server which has a log4j2.properties configuration and a cannot alter the format.

Regards

JF

**Comment by vy** [comment]
> Please had my vote on fixing this issue...

I will make sure this will be available in 2.14.1 which we will release in a couple of weeks.

> I am hoping to use this feature in a server which has a log4j2.properties configuration and a cannot alter the format.
> Currently using 2.14.0

Note that you don't necessarily need `eventTemplateAdditionalFields` to introduce custom fields. You can also use `eventTemplate` to pass along the entire event template including the custom fields. I hope this might serve you as a temporary workaround.

File: log4j-layout-template-json/src/test/java/org/apache/logging/log4j/layout/template/json/JsonTemplateLayoutAdditionalFieldTestProperties.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@LoggerContextSource("log4j2-properties-with-arrays.properties")
+class JsonTemplateLayoutAdditionalFieldTestProperties {
```
[the remark this answers is not in the document]
Comment (Fabricc) [reply log4j-layout-template-json/src/test/java/org/apache/logging/log4j/layout/template/json/JsonTemplateLayoutAdditionalFieldTestProperties.java:26]: Thank you for the feedback! I tried to stick to this example but I had some issues with the LoggerContextSource. It seems that every test gets only the first LoggerContextSource parsed. I tried to reconfigure the context and to programmatically change it but didn't work out. I also tried to use LoggerContextRule but that too didn't work as expect. Therefore I decided to create a common helper method and keep the classes separated but I have to admit, it doesn't look as concise as your suggestion. What do you think?

**Pull request merged**